front-end/src/pages/generatedPlaylist.js

```diff
@@ -116,15 +122,56 @@ const Playlist = (props) => {
       //only non-guests should be booted to the landing page
       return history.push("/");
     }
+    set_authentication(localStorage, axios);
 
     if (!isGuest && previousSongsRef.current === songs) {
       axios({
         method: "get",
-        url: `http://localhost:5000/groups/get_generated_playlist/${group_id}/${get_bearer(localStorage)}`,
+        url: `http://localhost:5000/groups/get_generated_playlist/${group_id}/${get_bearer(
+          localStorage
+        )}`,
       })
         .then((res) => {
-          setSongs(res.data.songs);
-          console.log(res.data.songs);
+          axios({
+            method: "get",
+            url: `https://api.spotify.com/v1/playlists/${location.state.generated_playlist_id}`,
+          })
+            .then((response) => {
+              setPlaylistAvatar(response.data.images[0].url);
```

Comment:
I think this may cause some problems if the playlist doesn't have an image? I tried deleting my playlist's image and it actually still brought up the original one, but I think perhaps spotify will eventually delete the image and this would cause an issue. Kind of a rare situation, but worth considering.